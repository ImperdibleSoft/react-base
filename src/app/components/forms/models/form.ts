import { FormState } from '../types/form';

export const getState = (): FormState => ({
  canBeLocked: true,
  hasPermissions: false,
  isEditEnabled: false,
});
