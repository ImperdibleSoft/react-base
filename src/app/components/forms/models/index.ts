import { FieldTypes, PersistedSmartForm, SmartFormState } from '../types';

export const TEXT: FieldTypes = 'text';
export const EMAIL: FieldTypes = 'email';
export const NUMBER: FieldTypes = 'number';
export const PASSWORD: FieldTypes = 'password';
export const TEXTAREA: FieldTypes = 'textarea';
export const SWITCH: FieldTypes = 'switch';
export const CHECKBOX: FieldTypes = 'checkbox';
export const DROPDOWN: FieldTypes = 'dropdown';
export const DATE: FieldTypes = 'date';
export const BUTTON: FieldTypes = 'button';
export const SUBMIT: FieldTypes = 'submit';
export const ARRAY: FieldTypes = 'array';

const initForm = (): PersistedSmartForm => {
  return {
    state: 'pristine',
    errors: [],
  };
};

export const getState = (): SmartFormState => ({
  form: initForm(),
});
