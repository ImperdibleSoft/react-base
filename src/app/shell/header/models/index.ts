import { HeaderMenu } from '../types';

export const noMenu: HeaderMenu = '';
export const modulesMenu: HeaderMenu = 'modules';
export const companiesMenu: HeaderMenu = 'companies';
export const userMenu: HeaderMenu = 'user';
