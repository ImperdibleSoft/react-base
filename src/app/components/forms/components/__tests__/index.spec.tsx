import * as React from 'react';
import { shallow } from 'enzyme';

import Field from '..';

import { BUTTON, CHECKBOX, DATE, DROPDOWN, PASSWORD, SUBMIT, SWITCH, TEXT, TEXTAREA } from '../../models';

describe('Field', () => {
  it('should render a valid button field', () => {
    const options = {
      id: 'button-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: BUTTON,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid checkbox field', () => {
    const options = {
      id: 'checkbox-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: CHECKBOX,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid date field', () => {
    const options = {
      id: 'date-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: DATE,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid dropdown field', () => {
    const options = {
      id: 'dropdown-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: DROPDOWN,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid password field', () => {
    const options = {
      id: 'password-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: PASSWORD,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid submit field', () => {
    const options = {
      id: 'submit-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: SUBMIT,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid switch field', () => {
    const options = {
      id: 'switch-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: SWITCH,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid text field', () => {
    const options = {
      id: 'text-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: TEXT,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });

  it('should render a valid textarea field', () => {
    const options = {
      id: 'textarea-field',
      label: 'Field label',
      onChange: () => {},
      onClick: () => {},
      onFocus: () => {},
      type: TEXTAREA,
    };

    const component = shallow(<Field options={options} />);

    expect(component).toMatchSnapshot();
  });
});
