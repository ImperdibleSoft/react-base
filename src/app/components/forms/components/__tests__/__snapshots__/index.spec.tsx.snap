// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should render a valid button field 1`] = `
<span
  key="button-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="button-field"
    name="button-field"
    render={[Function]}
  />
</span>
`;

exports[`Field should render a valid checkbox field 1`] = `
<span
  key="checkbox-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="checkbox-field"
    name="checkbox-field"
    render={[Function]}
  />
</span>
`;

exports[`Field should render a valid date field 1`] = `
<span
  key="date-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="date-field"
    name="date-field"
    render={[Function]}
  />
</span>
`;

exports[`Field should render a valid dropdown field 1`] = `
<span
  key="dropdown-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="dropdown-field"
    name="dropdown-field"
    render={[Function]}
  />
</span>
`;

exports[`Field should render a valid password field 1`] = `
<span
  key="password-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="password-field"
    name="password-field"
    render={[Function]}
  />
</span>
`;

exports[`Field should render a valid submit field 1`] = `
<span
  key="submit-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="submit-field"
    name="submit-field"
    render={[Function]}
  />
</span>
`;

exports[`Field should render a valid switch field 1`] = `
<span
  key="switch-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="switch-field"
    name="switch-field"
    render={[Function]}
  />
</span>
`;

exports[`Field should render a valid text field 1`] = `
<span
  key="text-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="text-field"
    name="text-field"
    render={[Function]}
  />
</span>
`;

exports[`Field should render a valid textarea field 1`] = `
<span
  key="textarea-field"
  style={
    Object {
      "display": "block",
      "maxWidth": undefined,
      "verticalAlign": "top",
    }
  }
>
  <Field
    id="textarea-field"
    name="textarea-field"
    render={[Function]}
  />
</span>
`;
